fmriprep/workflows/confounds.py

```diff
@@ -243,3 +264,211 @@ def reverse_order(inlist):
+
+
+def init_ica_aroma_wf(name='ica_aroma_wf', ignore_aroma_err=False):
+    '''
+    From: https://github.com/rhr-pruim/ICA-AROMA
+    Description:
+    ICA-AROMA (i.e. ‘ICA-based Automatic Removal Of Motion Artifacts’) concerns
+    a data-driven method to identify and remove motion-related independent
+    components from fMRI data.
+
+    Preconditions/Assumptions:
+    The input fmri bold file is in standard space
+    (for ease of interfacing with the original ICA-AROMA code)
+
+    Steps:
+    1) smooth data using SUSAN
+    2) run melodic outside of ICA_AROMA to generate the report
+    3) run ICA_AROMA
+    4) print identified motion components (aggressive) to tsv
+    5) pass classified_motion_ICs and melodic_mix for user to complete nonaggr denoising
+    '''
+    workflow = pe.Workflow(name=name)
+
+    inputnode = pe.Node(utility.IdentityInterface(fields=['epi_mni',
+                                                          'movpar_file',
+                                                          'epi_mask_mni']),
+                        name='inputnode')
+
+    outputnode = pe.Node(utility.IdentityInterface(
+        fields=['aroma_confounds', 'out_report',
+                'aroma_noise_ics', 'melodic_mix']), name='outputnode')
+
+    # helper function to get
+    # smoothing node (SUSAN)
+    # functions to help set SUSAN
+    def getbtthresh(medianval):
+        return 0.75 * medianval
+
+    def getusans_func(image, thresh):
+        return [tuple([image, thresh])]
+
+    calc_median_val = pe.Node(fsl.ImageStats(op_string='-k %s -p 50'),
+                              name='calc_median_val')
+
+    calc_epi_mean = pe.Node(fsl.MeanImage(),
+                            name='calc_epi_mean')
+
+    brightness_threshold = pe.Node(utility.Function(input_names=['medianval'],
+                                                    output_names=['thresh'],
+                                                    function=getbtthresh),
+                                   name='brightness_threshold')
+
+    getusans = pe.Node(utility.Function(input_names=['image', 'thresh'],
+                                        output_names=['usans'],
+                                        function=getusans_func),
+                       name='getusans')
+
+    smooth = pe.Node(fsl.SUSAN(fwhm=6.0),
+                     name='smooth')
+
+    # melodic node
+    melodic = pe.Node(nws.MELODICRPT(no_bet=True,
+                                     no_mm=True,
+                                     generate_report=True),
+                      name="melodic")
+
+    # ica_aroma node
+    ica_aroma = pe.Node(aroma.ICA_AROMA(denoise_type='no'),
+                        name='ica_aroma')
+
+    # extract the confound ICs from the results
+    ica_aroma_confound_extraction = pe.Node(utility.Function(input_names=['ica_out_dir',
+                                                             'ignore_aroma_err'],
```

Comment:
This can be done a little cleaner:

```Python
ica_aroma_confound_extraction = pe.Node(
    utility.Function(function=get_ica_confounds,
                     output_names=['aroma_confounds', 'aroma_noise_ics', 'melodic_mix']),
    name='ica_aroma_confound_extraction')
```

Other nodes can be cleaned up similarly. (I won't hold up merging on this, though. It's mostly an aesthetic thing.)